### DISCUSSION TITLE
Why supabase migrations up after supabase db diff

### DISCUSSION THREAD

**Discussion by gc-ft** [supabase/supabase#29032] [Questions]
After you use the supabase CLI to create a db diff for the local db vs migrations, a supabase migrations up will try to apply the latest migration script the CLI just generated to the local db. Is this not redundant? Should the CLI when creating a new migration via db diff not realise that the migration it just generated is already applied to the local db?

Am I missing something here with this behaviour?

Thanks,
Fabian

**Comment by jscheel** [comment]
I've run into the same thing. It doesn't make any sense to me.
